src/streaming/text/TextTracks.js

```diff
@@ -557,6 +565,41 @@ function TextTracks(config) {
         }
     }
 
+    function _handleCaptionEvents(currentItem, timeOffset, track, itemPos) {
+        let cue = new Cue(currentItem.start + timeOffset, currentItem.end + timeOffset, currentItem.data);
+        cue.cueId = currentItem.cueID || itemPos;
+        cue.data = currentItem.data;
+
+        cue.onenter = function () {
+            _fireCueEnter(cue);
+        }
+
+        cue.onexit = function () {
+            _fireCueExit(cue);
+        }
+
+        return cue;
+    }
+
+    function _fireCueEnter(cue) {
+        // This cleans up the VTT elements inside of the cue data.
+        const tmp = document.createElement('div');
+        tmp.innerHTML = cue.data;
```

Comment:
This should not be done in dash.js directly but rather in the application. We can dispatch cue.data and the application can remove VTT strings. Also don't we use all the formatting if we do this here?